src/main/java/com/example/java/maven/cardGamePeterPan/Game.java

```diff
@@ -0,0 +1,112 @@
+package com.example.java.maven.cardGamePeterPan;
+
+import java.util.ArrayList;
+
+public class Game {
+    private final UserInputProvider userInputProvider;
+    private final MessagePrinter messagePrinter;
+    private final ArrayList<Player> playerList;
+    private final Deck deck;
+
+    private final static String ASK_FOR_CARDS_TO_DISCARD= """
+            Which cards would you like to discard? 
+            Insert number telling which chosen card is, counting from left:""";
+    private final static String ASK_FOR_NEXT_CARD_TO_DISCARD = "Which else?";
+    private final static String WRONG_CARDS_TO_DISCARD = """
+            "You can't do that! 
+            You can discard two cards of the same colour or with the same figure!""";
+
+
+
+
+    public Game(UserInputProvider userInputProvider,
+                MessagePrinter messagePrinter,
+                ArrayList<Player> playerList,
+                Deck deck) {
+        this.userInputProvider = userInputProvider;
+        this.messagePrinter = messagePrinter;
+        this.playerList = playerList;
+        this.deck = deck;
+    }
+
+    public void dealCards() {
+        boolean areCards = true;
+        while (areCards) {
+            for (Player player : playerList) {
+                if (deck.getCards().size() > 0) {
+                    player.addCardToHand(deck.takeCardFromDeck());
+                }
+                if (deck.getCards().size() == 0) {
+                    areCards = false;
+                }
+            }
+        }
+    }
+
+    public void printPlayers() {
+        for (Player player : playerList) {
+            messagePrinter.printPlayer(player);
+        }
+    }
+
+    public void choseCardsToDiscard(Player player) {
+        messagePrinter.printPlayer(player);
+        messagePrinter.printMessage(ASK_FOR_CARDS_TO_DISCARD);
+        int firstCardToDiscardIndex = (getValidCardIndex(player));
+        messagePrinter.printMessage(ASK_FOR_NEXT_CARD_TO_DISCARD);
+        int secondCardToDiscardIndex = (getValidCardIndex(player));
+
+        discardCardsIfPossible(player, firstCardToDiscardIndex, secondCardToDiscardIndex);
+    }
+
+    private void discardCardsIfPossible(Player player, int firstCardToDiscardIndex, int secondCardToDiscardIndex) {
```

Comment:
this method will be simplified after applying previous suggestions

Reply:
Simplified